codegens/nodejs-axios/lib/axios.js

```diff
@@ -0,0 +1,225 @@
+    if (request.body.mode === 'file') {
+      request.addHeader({
+        key: 'Content-Type',
+        value: 'text/plain'
+      });
+    }
+    else if (request.body.mode === 'graphql') {
+      request.addHeader({
+        key: 'Content-Type',
+        value: 'application/json'
+      });
+    }
+  }
+
+  // The following code handles multiple files in the same formdata param.
+  // It removes the form data params where the src property is an array of filepath strings
+  // Splits that array into different form data params with src set as a single filepath string
+  if (request.body && request.body.mode === 'formdata') {
+    let formdata = request.body.formdata,
+      formdataArray = [];
+    formdata.members.forEach((param) => {
+      let key = param.key,
+        type = param.type,
+        disabled = param.disabled,
+        contentType = param.contentType;
+      // check if type is file or text
+      if (type === 'file') {
+        // if src is not of type string we check for array(multiple files)
+        if (typeof param.src !== 'string') {
+          // if src is an array(not empty), iterate over it and add files as separate form fields
+          if (Array.isArray(param.src) && param.src.length) {
+            param.src.forEach((filePath) => {
+              addFormParam(formdataArray, key, param.type, filePath, disabled, contentType);
+            });
+          }
+          // if src is not an array or string, or is an empty array, add a placeholder for file path(no files case)
+          else {
+            addFormParam(formdataArray, key, param.type, '/path/to/file', disabled, contentType);
+          }
+        }
+        // if src is string, directly add the param with src as filepath
+        else {
+          addFormParam(formdataArray, key, param.type, param.src, disabled, contentType);
+        }
+      }
+      // if type is text, directly add it to formdata array
+      else {
+        addFormParam(formdataArray, key, param.type, param.value, disabled, contentType);
+      }
+    });
+
+    request.body.update({
+      mode: 'formdata',
+      formdata: formdataArray
+    });
+  }
+  
+  body = request.body && request.body.toJSON();
+    
+  dataSnippet = parseRequest.parseBody(body, 
+                                    options.trimRequestBody,
+                                    indent,
+                                    request.headers.get('Content-Type'),
+                                    options.ES6_enabled);
+  snippet += dataSnippet+'\n';
+
+  configArray.push(indentString + `'method': '${request.method.toLowerCase()}'`);
+  configArray.push(indentString + `'url': '${sanitize(request.url.toString())}'`);  
+  
+  headers = parseRequest.parseHeader(request, indentString)
+  // https://github.com/axios/axios/issues/789#issuecomment-577177492
+  if (!_.isEmpty(body) && body.formdata ) {
+    // we can assume that data object is filled up
+    headers.push(`\n${indentString}...data.getHeaders()`);
```

Comment:
We don't really need to add a `\n` before the ...data.getHeaders(). I guess headers.join(,\n) will take care of it 🤔

Reply:
This is the snippet generated when request has some headers, and this formdata header is also added. The indentation looks weird, can you check on it once?
```
'headers': { 
      'my-sample-header': 'Lorem ipsum dolor sit amet', 
      'TEST': '@#$%^&*()', 
      'more': ',./\';[]}{\":?><|\\\\', 
  
  ...data.getHeaders()
  }
```

Reply:
done